### PR TITLE
docs: added release Apache APISIX 2.8 blog

### PR DESCRIPTION
**Pull request by Serendipity96**
Changes:

docs: added release Apache APISIX 2.8 blog

### CODE REVIEW

File: website/blog/2021-07-28-release-apache-apisix-2.8.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+从 2.8 版本开始，用户可以自定义 balancer。**这里的 balancer，是指最少连接数、轮询、一致性哈希等负载均衡器。**
+
+虽然 Apache APISIX 已经提供了丰富的 balancer，但是用户可能需要用的 balancer 是和业务紧密相关的 balancer，比如：需要考虑机房、可用区等等。支持自定义 balancer，用户可以编写自己的 balancer，并通过 `require("apisix.balancer.your_balancer")`  加载到程序中。
+
+通常自定义的 balancer 需要 node 提供 host/port 以外的数据，可以把数据放到 metadata 里面，举个例子：
+
+```JSON
+{
+    "nodes": [
+        { host = "0.0.0.0", port = 1980, weight = 1, metadata = {b = 1} }
```
Comment (juzhiyuan) [inline website/blog/2021-07-28-release-apache-apisix-2.8.md]:
```suggestion
        { host: "0.0.0.0", port: 1980, weight: 1, metadata: {b: 1} }
```

File: website/blog/2021-07-28-release-apache-apisix-2.8.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        "size": 4,
+        "idle_timeout": 8,
+        "requests": 16
+    }
+}
+```
+
+### 新体验：stream 代理功能增强
+
+在 2.8 版本中，把 ip-restriction 和 limit-conn 两个插件从 HTTP 部分的移植到了 stream 部分，这么做的好处是丰富网关在 stream 代理的功能，增加对上游服务的安全性保障。
```
Comment (tokers) [inline website/blog/2021-07-28-release-apache-apisix-2.8.md]: Add hyperlinks for these two plugins?

Comment (Serendipity96) [reply website/blog/2021-07-28-release-apache-apisix-2.8.md]: good idea！Can you tell me where are the hyperlinks？I'm not familiar with this part.

Comment (tokers) [reply website/blog/2021-07-28-release-apache-apisix-2.8.md]: * ip-restriction: http://apisix.apache.org/docs/apisix/plugins/ip-restriction/
* limit-conn: http://apisix.apache.org/docs/apisix/plugins/limit-conn/

File: website/blog/2021-07-28-release-apache-apisix-2.8.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
+---
+title: "Apache APISIX 2.8 正式发布,带来更多新功能！"
```
Comment (juzhiyuan) [inline website/blog/2021-07-28-release-apache-apisix-2.8.md]:
```suggestion
title: "Apache APISIX 2.8 正式发布，带来更多新功能！"
```

File: website/blog/2021-07-28-release-apache-apisix-2.8.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+author: spacewander
+authorURL: "https://github.com/spacewander"
+authorImageURL: "https://avatars.githubusercontent.com/u/4161644?v=4"
+---
+> [@spacewander](https://github.com/spacewander), Core developer of Apache APISIX from [Shenzhen Zhiliu Technology Co.](https://www.apiseven.com/)
+>
+<!--truncate-->
+
+Apache APISIX 2.8 版本正式发布！🎉 这个版本有 30+ 开发者参与，共提交了 100+ PR，支持了**1 个新功能，1 个新体验，2 个新插件，2 个新玩法**，快速阅读了解 2.8 版本的新特性吧！
```
Comment (juzhiyuan) [inline website/blog/2021-07-28-release-apache-apisix-2.8.md]:
```suggestion
Apache APISIX 2.8 版本正式发布！🎉 这个版本有 30+ 开发者参与，共提交了 100+ PR，支持了 **1 个新功能、1 个新体验、2 个新插件、2 个新玩法**，快去阅读了解 2.8 版本的新特性吧！
```

File: website/blog/2021-07-28-release-apache-apisix-2.8.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+            "comp_level": 1,
+            "vary": false
+        }
+    }
+}
+```
+
+### 新插件：ua-restriction
+
+ua-restriction 插件用于校验 User-Agent 是否处于黑白名单中，黑白名单是一个非常常见的需求，现在可以通过插件的方式启用了。
```
Comment (juzhiyuan) [inline website/blog/2021-07-28-release-apache-apisix-2.8.md]:
```suggestion
`ua-restriction` 插件用于校验 User-Agent 是否处于黑白名单中，黑白名单是一个非常常见的需求，现在可以通过插件的方式启用了。
```

File: website/blog/2021-07-28-release-apache-apisix-2.8.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+ua-restriction 配置举例：
```
Comment (juzhiyuan) [inline website/blog/2021-07-28-release-apache-apisix-2.8.md]:
```suggestion
`ua-restriction` 配置举例：
```

File: website/blog/2021-07-28-release-apache-apisix-2.8.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+从 2.8 版本开始，用户可以自定义 balancer。**这里的 balancer，是指最少连接数、轮询、一致性哈希等负载均衡器。**
+
+虽然 Apache APISIX 已经提供了丰富的 balancer，但是用户可能需要用的 balancer 是和业务紧密相关的 balancer，比如：需要考虑机房、可用区等等。支持自定义 balancer，用户可以编写自己的 balancer，并通过 `require("apisix.balancer.your_balancer")`  加载到程序中。
+
+通常自定义的 balancer 需要 node 提供 host/port 以外的数据，可以把数据放到 metadata 里面，举个例子：
+
+```JSON
+{
+    "nodes": [
+        { host: "0.0.0.0", port: 1980, weight: 1, metadata: {b: 1} }
```
Comment (juzhiyuan) [inline website/blog/2021-07-28-release-apache-apisix-2.8.md]:
```suggestion
        { "host": "0.0.0.0", "port": 1980, "weight": 1, "metadata": {"b": 1} }
```

File: website/blog/2021-07-28-release-apache-apisix-2.8.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+            "denylist": [
+                "my-bot1",
+                "(Baiduspider)/(\\d+)\\.(\\d+)"
+            ]
+        }
+    }
+}
+```
+
+### 新玩法：支持通过插件的方式实现 balancer
```
Comment (spacewander) [inline website/blog/2021-07-28-release-apache-apisix-2.8.md]: "通过插件的方式实现 balancer" is hard to distinguish from "自定义 balancer".
What about "支持在选择上游节点之后，通过插件的方式执行特定逻辑"?

Comment (Serendipity96) [reply website/blog/2021-07-28-release-apache-apisix-2.8.md:102]: New title is too long，it looks like a sentence. Do you have any other titles?
When we changed the title, do we need to rewrite the content below the title? I think we need.

Comment (Serendipity96) [reply website/blog/2021-07-28-release-apache-apisix-2.8.md]: New title is too long，it looks like a sentence. Do you have any other titles?
When we changed the title, do we need to rewrite the content below the title? I think we need.

Comment (spacewander) [reply website/blog/2021-07-28-release-apache-apisix-2.8.md]: What about "支持在选择上游节点之后执行自定义逻辑"?

Comment (Serendipity96) [reply website/blog/2021-07-28-release-apache-apisix-2.8.md]: The title is good. If we change the title, the content below need to rewrite.  The content now introduces how to implementation of balancer by plug-in. It is different the title describe. Which one do you introduce to others, plugin or customized logic?

**Review by spacewander — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**